src/test/addingToFavourites.test.tsx

```diff
@@ -1,63 +1,69 @@
+import "@testing-library/jest-dom";
 import { fireEvent, screen, waitFor } from "@testing-library/react";
+import BookCard from "components/BookCard";
+import type { Book } from "src/types";
 import { render } from "test/test-utils";
-import "@testing-library/jest-dom";
-import Book from "components/Book";
 
-const dummyBook = {
+const dummyBook: Book = {
   title: "The Great Gatsby",
   authors: [{ name: "F. Scott Fitzgerald" }],
   bookshelves: ["Best Books Ever"],
-  formats: {},
+  formats: {
+    "text/html": "https://www.gutenberg.org/ebooks/64317",
+    "image/jpeg":
+      "https://www.gutenberg.org/cache/epub/64317/pg64317.cover.medium.jpg",
+  },
   languages: ["en"],
   subjects: ["fiction"],
   translators: [],
   id: 5,
 };
 
-test("test changing star when adding/deleting to/from favourites", async () => {
+it("test changing star when adding/deleting to/from favourites", async () => {
   // ARRANGE
-  render(<Book book={dummyBook} />);
+  render(<BookCard book={dummyBook} />);
 
   // ACT
+  const notFavouriteStar = await screen.findByTestId("not-favourite");
   await waitFor(() => {
-    const notFavouriteStar = screen.getByTestId("not-favourite");
     expect(notFavouriteStar).toBeInTheDocument();
-    fireEvent.click(notFavouriteStar);
   });
+  fireEvent.click(notFavouriteStar);
+  const favouriteStar = await screen.findByTestId("favourite");
   await waitFor(() => {
-    const favouriteStar = screen.getByTestId("favourite");
     expect(favouriteStar).toBeInTheDocument();
-    fireEvent.click(favouriteStar);
   });
```

Comment:
to nie jest w ogóle potrzebne 😭 (ten waitFor), wyżej `findBy` Ci nie przejdzie, dopóki nie znajdzie tego w dokumencie